tests/integration/components/o-s-s/checkbox-test.ts

```diff
@@ -0,0 +1,60 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { click, render, setupOnerror } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import sinon from 'sinon';
+
+module('Integration | Component | o-s-s/checkbox', function (hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function () {
+    this.onChange = () => {};
+  });
+
+  test('it throws an error if checked argument is missing', async function (assert) {
+    setupOnerror((error: Error) => {
+      assert.equal(error.message, '[component][OSS::Checkbox] Boolean @checked argument is mandatory.');
+    });
+    await render(hbs`<OSS::Checkbox />`);
+  });
+
+  test('it throws an error if checked argument is missing', async function (assert) {
+    setupOnerror((error: Error) => {
+      assert.equal(error.message, '[component][OSS::Checkbox] @onChange argument is mandatory.');
+    });
+
+    await render(hbs`<OSS::Checkbox @checked={{true}} />`);
+  });
+
+  test('it renders correctly', async function (assert: Assert) {
+    this.checked = false;
+
+    await render(hbs`<OSS::Checkbox @checked={{this.checked}} @onChange={{this.onChange}} />`);
+
+    assert.dom('.upf-checkbox input').isNotChecked();
+    this.set('checked', true);
+    assert.dom('.upf-checkbox input').isChecked();
```

Comment:
How about moving this into a separate test ? Since you are actually testing that when the parent updates the state of the boolean, it is indeed applied ?

Also, might be overkill but, don't you want to check that the custom "upf-fake-checkbox" exists ?